packages/store/addon/-private/system/weak-cache.ts

```diff
@@ -0,0 +1,70 @@
+import { DEBUG } from '@glimmer/env';
+
+import { addSymbol, symbol } from '../ts-interfaces/utils/symbol';
+
+interface WeakCache<K extends object, V> {
+  _symbol?: Symbol | string;
+  _fieldName?: string;
+  _generator?: (key: K) => V;
+  _expectMsg?: (key: K) => string;
+}
+
+class WeakCache<K extends object, V> {
```

Comment:
Also, authoring this as a subclass removes the need for all the proxy style methods, and inherit all the existing `WeakMap` typings.

```js
class NamedWeakMap extends WeakMap {
  private label: string;
  constructor(label) {
    super();
    this.label = label;
  }

  // ....
}
```

Reply:
Implemented